packages/contracts/contracts/generator/scripty-support/ETHFSFileStorage.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MIT
```

Comment:
@ryley-o There's a new version of ethFS (currently the only version on sepolia) which uses a different File.sol/IFile.sol so we needed to deploy this ourselves. That's also why I've added the ethfs directory here. For prod we shouldn't need to deploy anything ourselves for scripty/ethfs so we may want to remove this.

Reply:
SGTM - Is this a forked version of scripty? Wasn't quite sure why this file has to exist?
A little readme in here and ethfs directories that points to the source repos to give credit probably makes sense. IIUC, we cannot just import ethfs, because there are no official release of it?
Also, IIUC, scripty is relying on ethfs's FileStore interface, which is owned by an EOA on mainnet. I think it may be worth noting that dependency/risk in the natspec of GenArt721GeneratorV0.sol, yeah? (I think it is an acceptable risk for now, fwiw, just want to document it).

Reply:
This is pretty much just for the gunzip script. Scripty allows you to pass in a contract address and name for script tags like [this](https://int-art.gitbook.io/scripty.sol-v2/building-html/getting-html-on-chain#generating-script-tag-with-its-content-fetched-from-ethfs) but the address needs to implement [IContractScript](https://github.com/intartnft/scripty.sol/blob/main/contracts/scripty/interfaces/IContractScript.sol). They have an IContractScript [wrapper](https://github.com/intartnft/scripty.sol/blob/main/contracts/scripty/externalStorage/ethfs/ETHFSFileStorage.sol) around the ethfs FileStore contract so that it's possible to use in this way but the wrapper they have uses an older version of ethfs' File and IFileStore which is not compatible with the new version of ethfs (the only one deployed to sepolia). This file is the same as the scripty ETHFSFileStorage but imports the updated ethfs files for compatibility with the new version.

Reply:
Also, ethfs isn't available as an npm package so we need the supporting files manually added to our repo.